### PR TITLE
Feat/drop

### PR DESCRIPTION
**Pull request by streamich**

### CODE REVIEW

File: src/DropArea/index.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import {h, noop} from '../util';
+
+export interface IDropAreaProps {
+  onFiles?: (files: File[], event?) => void;
+  onText?: (text: string, event?) => void;
+  onUri?: (url: string, event?) => void;
+}
+
+export interface IDropAreaState {
+}
```
Comment (carpasse) [inline src/DropArea/index.ts]: empty state?

File: src/DropArea/index.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  };
+
+  mounted = true;
+
+  coponentWillUnmount () {
+    this.mounted = false;
+  }
+
+  onDragOver = (originalDragOver) => (event) => {
+    (originalDragOver || noop)(event);
```
Comment (carpasse) [inline src/DropArea/index.ts]: why don't you make `noop` the default value of `originalDragOver` param?

File: src/DropArea/index.ts (hunk 1 above)
Comment (streamich) [reply src/DropArea/index.ts]: Yes, it only uses life-cycle methods.

File: src/DropArea/index.ts (hunk 2 above)
Comment (streamich) [reply src/DropArea/index.ts]: Nice idea, will do.

**Pull request merged**